pgn_downloader/main.py

```diff
@@ -43,17 +33,28 @@ def download_pgn(username: str, color: str):
         print(f"Downloading games from {month_url[-7:]}", end="\r")
         games = requests.get(month_url).json()["games"]
         for game in games:
-            # download only games with requested color
-            if game[f"{color}"]["username"] == username:
-                
-                try:
-                    pgn += game["pgn"] + "\n\n"
-                    nr_games += 1
-                except KeyError:
-                    pass
-            else:
-                continue
 
+            # TODO: How to write this in an elegant way without copying code? Too many if-statements...
```

Comment:
True.
One way would be to use a `filter_games()` function, which takes color and maybe additional filtering parameters, which just decides if the game should be saved or not.
The check if the pgn even exists can also be enclosed in this.